vscode_extension/src/aacExecutableWrapper.ts

```diff
@@ -0,0 +1,123 @@
+
+    const quickPickOptions: QuickPickOptions = {
+        canPickMany: false,
+    };
+
+    window.showQuickPick(availableCommands, quickPickOptions).then(commandName => {
+        if(commandName) {
+            getAacCommandArgs(commandName).then(async (commandArguments) => {
+                await getCommandArgUserInput(commandArguments);
+                execAacShellCommand(commandName, commandArguments).then(output => {
+                    outputChannel.appendLine(output);
+                    outputChannel.show();
+                });
+            });
+        }
+    });
+}
+
+async function getCommandArgUserInput(commandArguments: CommandArgument[]) {
+
+    let unansweredCommandArguments: CommandArgument[] = commandArguments.filter(argument => { return (argument.userResponse === undefined); } );
+
+    if (unansweredCommandArguments.length > 0) {
+        let argumentToPromptFor = unansweredCommandArguments[0];
+        const dialogBoxOptions: OpenDialogOptions = {
+            title: argumentToPromptFor.name,
+        };
+        const inputBoxOptions: InputBoxOptions = {
+            title: argumentToPromptFor.name,
+            prompt: argumentToPromptFor.description,
+        };
+
+        if (argumentToPromptFor.description.toLowerCase().includes("path")){
+            let fileUri: Uri[] | undefined = await window.showOpenDialog(inputBoxOptions);
+            if(fileUri)
+                {argumentToPromptFor.userResponse = fileUri[0]?.path;}
+        } else {
+            argumentToPromptFor.userResponse = await window.showInputBox(inputBoxOptions);
+        }
+    } else if(unansweredCommandArguments.length > 1) {
+        getCommandArgUserInput(commandArguments);
+    }
+}
+
+async function getAacCommandNames(): Promise<string[]> {
+    const aacHelpOutput = await execAacShellCommand("-h");
+    return parseTaskNamesFromHelpCommand(aacHelpOutput);
+}
+
+/**
+ * @returns list of Command arguments
+ */
+async function getAacCommandArgs(aacCommandName: string): Promise<CommandArgument[]> {
+    const aacHelpOutput = await execAacShellCommand(`${aacCommandName} -h`);
+    return parseTaskArgsFromHelpCommand(aacHelpOutput);
+}
+
+async function execAacShellCommand(command: string, commandArgs: CommandArgument[] = []): Promise<string> {
+
+    let result: string = "";
+
+    let commandArgsArray = ["aac", command, ...(commandArgs.map(argument => argument.userResponse))];
+    try {
+        const { stdout, stderr } = await execShell(commandArgsArray.join(" "), {});
+        const stringOutput = stderr.length > 0 ? stderr : stdout;
+        return stringOutput;
+
+    } catch (error: any) {
+        let errorMessage = error.stderr || error.stdout || "";
+
+        outputChannel.appendLine(`Failed to execute AaC command:\n${errorMessage}`);
+        outputChannel.show(true);
+        throw error;
+    }
+}
+
+function parseTaskNamesFromHelpCommand(aacHelpOutput: string): string[] {
+
+    const regExp = /{(.*)}/;
```

Comment:
I don't think this will work as expected, we need to use a non-greedy version of this regular expression. See [here](https://www.typescriptlang.org/play?#code/MYewdgzgLgBAFgUwDYAcDyBXKKswLwwBEGEAhgOYIBcMppwMA2gLRwC6AOgFAy9-8wA3gDdSSAJYATUlAQAaYQgBOEceDl1gzUEoTMIKBMDnRSSqMyQGThraInTZcnAFsk2kC5TgEYKM4w3fQQARwxfYHlXdxAAIwArI39KMGZyRCUAa3FU2IQ4UmE1FTkU5kkEVXJUyRBjMpQlECg4jAAzUt9mFCQMchzO1PiIcABfbgFJgDoZ7m5vVSg1MDFaJXJA3ygIKgmheykZeUUVZY16D119WxMoMwsrFBsjZgPHKMD3UC8fPwCgiChcJgSL-GIJJKDNIZbK5fKFYoQKEVKo1OpQxrNVodBq9fpgKHDMZ7XhvI6TCkCACikAwuggMCgiBgAEFSABhGAAT1Ibhg4gZbxghiUjOZbM5OgQMAMRhJMBOqnAlJVvEw2CwDJZSmAcHEsmAUDpelIEGY7JAFRgAAUuUzlSh6JkKNLFct5ZpLnpZQxVbxrUoctsxdKJTAXJbkDAUcBA7EcuQYFLWRzozJSIyuYYGaQwJIYL5NkoZMsIPLTOZLAY-ZMAMp3cwhlOcgAyuY2LpgteUJxtTRaoCQMAAFC3a9aAJQynvKcu2V5iQ6yGswABqi-eDJ9jOLkQZAHd9XqwE2wxGKkgpvLoh4fmAtjWLWATsHc83w5Ghy1aDakLnYAAqgAsi2SaeN495+NG4gUMWLhXjwfA3oCYQRNKj7gC+OYnmGJBGKa0rfhm1p-lBwGgShwKRNBsG8gh-A3nEiSGiuvBPlhtA4am55RkRv7-jA5EwExSQ0eQcH0XwZTpMosLMHkBRFCAKgCAA4r4yjkqpMI5DAbQIDIxp6eISCVHpTQuKyOpwMwprmpGH4XjACkIspMqRCsgYgBAkm8GUKLiNU5R1DW6n3sWy4ZrWXLQAglkACKVIFJ7xXUmxQW0FlWbq+pJMatlmhaVo8VYvkwA0-bYqxYWacumItLE7ThpUZCUAymWeNlNl2UV0olT58q4n0umsTANURWZb5hj0w0nh1llTdxn7CW0TYzfig1dESJ6jWxmHKK+XGciVjIgDAABStZoAAcnMXAgCgSzgKsZgbC4Ww7HsrByDAzCsMgKC7bwEBwCA+5igK8AA81ECtdKub5ggAAe+qEAA3FwXBtBgIJPSelBQAAKqamQQMOug0AASgg5BUkjKBToIiG8KAkCwCs70MgQuhTMjRjDogqDqjgUAThj-DiKtw4c5UjPMwIuhGkoJ4yxAjAAIxsFMBgSFAw6EHIhBi-Lozy4rdInowbBcKbXAAPR2zAJAJjAYDgGkugIJIXIwLolBI1wrMjKZUxICA5DDgTxMQKTw524Iw5TAAVAA-BOox2xOxv247ztgImbtKC4qzDuIvNTOVnve1OfvI4H4DBwgofh5HCBEyTZPx4nSfp5nxtAA) for the difference.

```suggestion
    const regExp = /{(.*?)}/;
```